crypto/conf/conf_lib.c

```diff
@@ -292,10 +292,13 @@ int NCONF_get_number_e(const CONF *conf, const char *group, const char *name,
     if (str == NULL)
         return 0;
 
-    for (*result = 0; conf->meth->is_number(conf, *str);) {
-        *result = (*result) * 10 + conf->meth->to_int(conf, *str);
-        str++;
-    }
+    if (conf == NULL)
+        *result = strtol(str, &str, 10);
```

Comment:
One has to recognize that this is actually not equivalent replacement. Because strtol pays attention to `-`, is saturating and sets errno. One can of course argue that it's the right thing to do. [Maybe modulo errno? Maybe it would be appropriate to save-restore it?] But then this calls for question if below code is right. One should note that it's actually prone to undefined behaviour. Because signed arithmetics that carries into sign bit is declared undefined. It might be appropriate to perform unsigned operations, pay attention to carry, then finally perform check for LONG_MAX and saturate? And maybe it should pay attention to `-`. Alternatively it might be more appropriate to use strtoul with cast as above... Or right thing to do might be to have called passing default method...